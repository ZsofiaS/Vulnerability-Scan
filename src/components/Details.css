.Details {
  border: 1px solid gainsboro;
  margin: 2rem 10rem;
  padding: 1rem 2rem;
  text-align: left;
  border-radius: 5px;
}
