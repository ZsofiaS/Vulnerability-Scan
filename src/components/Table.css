.table {
  table-layout: fixed;
  width: 100%;
  border-collapse: collapse;
  text-align: center;
}
th, td {
  padding: 0.5rem 0.3rem;
}
.header {
  background-color: #F4F6F6;
}
