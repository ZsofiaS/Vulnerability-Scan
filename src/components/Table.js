import React from 'react';
import data from '../data/scan';
import TableRow from './TableRow';

const Table = () => {

  const renderTable = () => {
    return data.scan.vulnerabilities.map((vulnerability, i) => {
      const { id, severity, name, description, solution, references, cvssBaseScore, affectedAssets } = vulnerability;
      return (
        <TableRow
          key={i}
          severity={severity}
          name={name}
          description={description}
          solution={solution}
          references={references}
          cvssBaseScore={cvssBaseScore}
          affectedAssets={affectedAssets}
          renderAssets={renderAssets(affectedAssets)}
        />
      )
    })
  }

  const renderAssets = (assets) => {
    return assets.map((asset, i) => {
      let item = data.scan.assets.find(obj => obj.id == asset);
      return (
        <span key={i}>{item.description} </span>
      )
    })
  }

  return (
    <table>
      <thead>
        <tr>
          <th>Severity</th>
          <th>Issue</th>
          {/*
          <th>Summary</th>
          <th>Solution</th>
          <th>References</th>
          */}
          <th>CVSS Base Score</th>
          <th>Affected Assets</th>
        </tr>
      </thead>
      <tbody>
        {renderTable()}
      </tbody>
    </table>
  )
}

export default Table;
