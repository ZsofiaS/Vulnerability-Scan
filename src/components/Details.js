import React from 'react';
import Table from './Table';
import './Details.css';

const Details = () => {

  return (
    <div className="Details">
      <h1>Vulnerabilities found</h1>
      <Table />
    </div>
  )
}

export default Details;
