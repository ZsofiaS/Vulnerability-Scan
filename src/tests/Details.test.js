import React from 'react';
import { shallow, mount } from 'enzyme';
import Details from '../components/Details';
import Table from '../components/Table';
import toJson from 'enzyme-to-json';

describe('renders Details component', () => {
  it('renders without crashing', () => {
    shallow(<Details />);
  });
  it('renders header', () => {
    const wrapper = shallow(<Details />);
    const header = (<h2>Vulnerabilities found</h2>);
    expect(wrapper.contains(header)).toEqual(true);
  });
  it('renders Table component', () => {
    shallow(<Table />);
  })
});
