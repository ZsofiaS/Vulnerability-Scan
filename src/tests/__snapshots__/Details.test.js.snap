// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Details snapshot 1`] = `
<div
  className="details"
>
  <h2>
    Vulnerabilities found
  </h2>
  <Table />
</div>
`;
